### PR TITLE
Brings Heretics Up To Date

### PR DESCRIPTION
**Pull request by eeSPee**
## About The Pull Request

Brings heretics code up to date with these major changes:
* Tears in reality no longer inflict brain damage when analyzed, instead just give a bad moodlet
* Revised reality tears generation
* Tears in reality appear after a time, and slowly disappear after 15 minutes
* Heretic effects (runes, tears) are invisible to silicons
* Names and descriptions have been revised, thanks to @EOBGames
* Heart now offers a pick between 3 targets
* Doubled the charges sacrifices offer
* Heretics marks area work differently; marks are applied by grasp, and triggered with the blade
* Ash grasp blinds instead of yeeting
* Rust spread datum is revised/optimized

Up to date with the PRs:
* https://github.com/tgstation/tgstation/commit/283d140ca08c1cdc2f6d9f46e9458c29e270b905#diff-f13197ae0925553052a7e70a8cd27e9a97baf2b54f00182a36a7b4c7a25cfbc3
* https://github.com/tgstation/tgstation/commit/3190cc36038874803bf198f06b2eb0f48bf2724a#diff-52a3ee8ccf1b6f2e59a60d8ff609db64bd4c65424592664c508d0f657bd0cae4
* https://github.com/tgstation/tgstation/commit/7c123a2fe27206f8af35b8455dab7e7ba24b7156#diff-0bcd93446e8e6938a1603b628e91bbf6be72214112c70198f95f48bae9b44a67
* https://github.com/tgstation/tgstation/pull/54372
* https://github.com/tgstation/tgstation/pull/54371
* https://github.com/tgstation/tgstation/pull/54967
* https://github.com/tgstation/tgstation/pull/54475
* https://github.com/tgstation/tgstation/pull/53414
* https://github.com/tgstation/tgstation/pull/53396
* https://github.com/tgstation/tgstation/pull/52680

## Why It's Good For The Game

Heretics is out of date and tg made many QoL changes that improves performance and how heretics work.

## Changelog
:cl: EdgeLordExe, EOBGames, ShivCalev, Rohesie, Tad Hardesty, ATHATH, and eeSPee
tweak: Tears in reality no longer inflict brain damage when analyzed, instead just give a bad moodlet
tweak: Revised reality tears generation
tweak: Tears in reality appear after a time, and slowly disappear after 15 minutes
tweak: Coders rule, jannies drool
balance: Heretic effects (runes, tears) are invisible to silicons
spellcheck: Names and descriptions have been revised, thanks to @EOBGames
balance:  Heart now offers a pick between 3 targets
balance: Doubled the charges sacrifices offer
balance: Heretics marks area work differently; marks are applied by grasp, and triggered with the blade
balance: Ash grasp blinds instead of knockback
tweak: Rust spread datum is revised/optimized
/:cl:

### CODE REVIEW

File: code/modules/antagonists/eldritch_cult/eldritch_effects.dm [hunk 1]
```diff
@@ -243,47 +231,27 @@
...
 	///Tracked image
 	var/image/img
+	invisibility = INVISIBILITY_OBSERVER
 
 /obj/effect/reality_smash/Initialize()
 	. = ..()
-	img = image(icon, src, image_state, OBJ_LAYER)
+	//var/img = image(icon, src, "reality_smash", OBJ_LAYER)
 	generate_name()
+	// This was on TG but I'm not sure what the bee equivalent is --- add_alt_appearance(/datum/atom_hud/alternate_appearance/basic/heretics,"influence",img)
```
Comment (PowerfulBacon) [inline code/modules/antagonists/eldritch_cult/eldritch_effects.dm]: this is the shit that makes it invisible to non-heretics I'm pretty sure

Comment (eeSPee) [reply code/modules/antagonists/eldritch_cult/eldritch_effects.dm]: I'm not sure why I commented it out. I think it failed my build or whatever.

**Comment by eeSPee** [comment]
Sorry for so many Commits; dream maker is acting up and won't let me compile.

File: code/modules/antagonists/eldritch_cult/eldritch_effects.dm (hunk 1 above)
Comment (eeSPee) [reply code/modules/antagonists/eldritch_cult/eldritch_effects.dm]: Ah; it should work now properly.

**Comment by heepox** [comment]
so much yes.

**Comment by bloons3** [comment]
@eeSPee 

This doesn't prevent silicons from seeing heretic runes/pierced realities?

https://github.com/tgstation/tgstation/pull/54372

Missing this PR

**Comment by Fronsis** [comment]
Why are they invisible to sillicons?

**Comment by bloons3** [comment]
> Why are they invisible to sillicons?

https://github.com/tgstation/tgstation/pull/54372

Cult runes are obscured from silicons as well, so consistency

File: code/datums/mood_events/generic_negative_events.dm [hunk 2]
```diff
@@ -217,3 +217,8 @@
 
 /datum/mood_event/nanite_sadness/add_effects(message)
 	description = "<span class='warning robot'>+++++++[message]+++++++</span>\n"
+
+/datum/mood_event/gates_of_mansus
+	description = "<span class='boldwarning'>I HAD A GLIMPSE OF THE HORROR BEYOND THIS WORLD. REALITY UNCOILED BEFORE MY EYES!</span>\n"
+	mood_change = -25
+	timeout = 4 MINUTES
```
Comment (park66665) [inline code/datums/mood_events/generic_negative_events.dm]:
```suggestion
```

We already have `/datum/mood_event/gates_of_mansus`. It is about twenty lines above.

File: code/modules/antagonists/eldritch_cult/eldritch_effects.dm [hunk 3]
```diff
@@ -67,6 +73,7 @@
 				if(is_type_in_list(local_atom_in_range,local_required_atom_list))
 					selected_atoms |= local_atom_in_range
 					local_required_atoms -= list(local_required_atom_list)
+					break
```
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_effects.dm:76]: This change is from tgstation/tgstation#54371. Better credit it.

File: code/modules/antagonists/eldritch_cult/eldritch_antag.dm [hunk 4]
```diff
@@ -52,11 +54,16 @@
...
 	STOP_PROCESSING(SSprocessing,src)
 
+	on_death()
 	return ..()
 
+/datum/antagonist/heretic/proc/on_death()
+
+	for(var/X in researched_knowledge)
+		var/datum/eldritch_knowledge/EK = researched_knowledge[X]
+		EK.on_death(owner.current)
```
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_antag.dm]: `on_death` part is from tgstation/tgstation#54252

File: code/modules/antagonists/eldritch_cult/eldritch_effects.dm [hunk 5]
```diff
@@ -90,59 +97,42 @@
...
-  * Stupid fucking list holder, DONT create new ones, it will break the game, this is automnatically created whenever eldritch cultists are created.
-  *
-  * Tracks relevant data, generates relevant data, useful tool
-  */
+ * #Reality smash tracker
+ *
+ * Stupid fucking list holder, DONT create new ones, it will break the game, this is automnatically created whenever eldritch cultists are created.
+ *
+ * Tracks relevant data, generates relevant data, useful tool
+ */
```
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_effects.dm:116]: This set of changes are from tgstation/tgstation#54967. And better change the comment.

File: code/modules/antagonists/eldritch_cult/eldritch_effects.dm [hunk 6]
```diff
@@ -198,42 +186,43 @@
...
-			head.dismember()
-			qdel(head)
-		else
-			human_user.gib()
-
+		human_user.gib()
 		var/datum/effect_system/reagents_explosion/explosion = new()
-		explosion.set_up(1, get_turf(human_user), 1, 0)
+		explosion.set_up(1, get_turf(human_user), TRUE, 0)
 		explosion.start()
```
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_effects.dm:211]: This set of changes also includes changes from tgstation/tgstation#54475.

File: code/modules/antagonists/eldritch_cult/eldritch_effects.dm [hunk 7]
```diff
@@ -198,42 +186,43 @@
...
 
 /obj/effect/broken_illusion/examine(mob/user)
-	if(!IS_HERETIC(user) && ishuman(user))
-		var/mob/living/carbon/human/human_user = user
-		to_chat(human_user,"<span class='warning'>Your brain hurts when you look at this!</span>")
-		human_user.adjustOrganLoss(ORGAN_SLOT_BRAIN,10)
 	. = ..()
```
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_effects.dm:214]: Movement of `. = ..()` is from tgstation/tgstation#53414

File: code/modules/antagonists/eldritch_cult/eldritch_effects.dm [hunk 8]
```diff
@@ -243,47 +232,27 @@
 	var/list/minds = list()
 	///Tracked image
 	var/image/img
+	invisibility = INVISIBILITY_OBSERVER
 
 /obj/effect/reality_smash/Initialize()
 	. = ..()
-	img = image(icon, src, image_state, OBJ_LAYER)
+	//var/img = image(icon, src, "reality_smash", OBJ_LAYER)
```
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_effects.dm]: You didn't properly port tgstation/tgstation#54967.

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm [hunk 9]
```diff
@@ -25,6 +24,10 @@
 	var/list/result_atoms = list()
 	///What path is this on defaults to "Side"
 	var/route = PATH_SIDE
+	///Cast delay when using the rune
+	var/cast_time = 10	//short delay so you can cancel?
+	//Increases the total amount of followers
+	var/followers_increment = 0
```
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm]: Those variables are not used in this PR. Are you going to use them in future PRs?

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm [hunk 10]
```diff
@@ -144,15 +169,13 @@
 	var/list/compiled_list = list()
 
 	for(var/H in GLOB.carbon_list)
-		if(!ishuman(H))
-			continue
```
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm:148]: You should restore these lines. We don't have `GLOB.human_list`

File: code/modules/antagonists/eldritch_cult/knowledge/ash_lore.dm [hunk 11]
```diff
@@ -28,44 +28,39 @@
...
 /datum/eldritch_knowledge/ashen_grasp/on_mansus_grasp(atom/target, mob/user, proximity_flag, click_parameters)
 	. = ..()
-	if(!iscarbon(target))
+	var/mob/living/carbon/C = target
+	if(!istype(C))
 		return
+	to_chat(C, "<span class='danger'>Your eyes burn horrifically!</span>") //pocket sand! also, this is the message that changeling blind stings use, and no, I'm not ashamed about reusing it
+	C.become_nearsighted(EYE_DAMAGE)
+	C.blind_eyes(5)
+	C.blur_eyes(10)
```
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/knowledge/ash_lore.dm:37]: This change is from tgstation/tgstation#53396

File: code/modules/antagonists/eldritch_cult/knowledge/rust_lore.dm [hunk 12]
```diff
@@ -172,27 +157,38 @@
...
-		for(var/turf/T in range(1,X))
-			if(T in turfs)
-				continue
-			if(is_type_in_typecache(T,blacklisted_turfs))
-				continue
-			edge_turfs += T
+		for(var/line_turfie_owo in getline(turfie,centre))
+			if(get_dist(turfie,line_turfie_owo) <= 1)
+				edge_turfs += turfie
+		CHECK_TICK
```
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/knowledge/rust_lore.dm]: This set of changes is from tgstation/tgstation#52680

**Review by park66665 — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/eldritch_cult/eldritch_antag.dm (hunk 4 above)
Comment (eeSPee) [reply code/modules/antagonists/eldritch_cult/eldritch_antag.dm]: Thanks. I missed this one.

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm (hunk 10 above)
Comment (eeSPee) [reply code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm:148]: No need; now checks if istype isntead of ishuman, see the lines below.

Comment (park66665) [reply code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm:148]: Ohhhhh there it is. I missed that one.

**Comment by eeSPee** [comment]
Thanks! Fixed the code and cited the proper PRs.

File: code/modules/antagonists/eldritch_cult/eldritch_effects.dm [hunk 13]
```diff
@@ -198,92 +184,66 @@
...
 
 /obj/effect/broken_illusion/examine(mob/user)
-	if(!IS_HERETIC(user) && ishuman(user))
-		var/mob/living/carbon/human/human_user = user
-		to_chat(human_user,"<span class='warning'>Your brain hurts when you look at this!</span>")
-		human_user.adjustOrganLoss(ORGAN_SLOT_BRAIN,10)
 	. = ..()
-
+	var/mob/living/carbon/human/human_user = user
+	if(istype(user) && !IS_HERETIC(user) && !IS_HERETIC_MONSTER(user))
```
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_effects.dm]:
```suggestion
	if(istype(human_user) && !IS_HERETIC(human_user) && !IS_HERETIC_MONSTER(human_user))
```

File: code/modules/antagonists/eldritch_cult/eldritch_effects.dm [hunk 14]
```diff
@@ -90,104 +97,83 @@
...
+	addtimer(CALLBACK(src,.proc/dissipate),15 MINUTES)
+
+///Makes this obj appear out of nothing
+/obj/effect/broken_illusion/proc/show_presence()
+	animate(src,alpha = 255,time = 15 SECONDS)
+	
+/obj/effect/broken_illusion/proc/dissipate()
+	animate(src,alpha = 0,time = 2 MINUTES)
+	sleep(2 MINUTES)
+	qdel(src)
```
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_effects.dm]:
```suggestion
	QDEL_IN(src, 2 MINUTES)
```

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm [hunk 15]
```diff
@@ -128,6 +140,7 @@
 /datum/eldritch_knowledge/curse
 	var/timer = 5 MINUTES
 	var/list/fingerprints = list()
+	var/list/dna = list()
```
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm]: This `dna` list is not used. Its usage isn't implemented in tg either. So you have a choice. You implement it, or you don't port the remnants of dna check.

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm [hunk 16]
```diff
@@ -260,28 +271,35 @@
 				if(!istype(X,/obj/item/forbidden_book))
 					continue
 				var/obj/item/forbidden_book/FB = X
-				FB.charge++
+				FB.charge += 2
```
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm:273]: Remind you, this change will conflict with #3451

Comment (eeSPee) [reply code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm:273]: Yes, #3451 should be implemented after this one.

This shouldn't be a problem.

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm (hunk 15 above)
Comment (eeSPee) [reply code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm]: I think this is a leftover of some sorts.

**Comment by utilbird** [comment]
Ah right, should probably merge this before we merge any other heretic PRs

**Comment by heepox** [comment]
i wonder if this will have to be atomized for anything to be merged...

Combat Tweak
- balance: Ash grasp blinds instead of knockback
- Heretics marks area work differently; marks are applied by grasp, and triggered with the blade

Tweaks to Gamemode in general
- tweak: Rust spread datum is revised/optimized
- tweak: Tears in reality no longer inflict brain damage when analyzed, instead just give a bad moodlet
- tweak: Revised reality tears generation
- tweak: Tears in reality appear after a time, and slowly disappear after 15 minutes
- balance: Heretic effects (runes, tears) are invisible to silicons
- spellcheck: Names and descriptions have been revised, thanks to @EOBGames

Heart Changes
- Heart now offers a pick between 3 targets
- Doubled the charges sacrifices offer

**Review by utilbird — DISMISSED** [review]

**Comment by eeSPee** [comment]
> i wonder if this will have to be atomized for anything to be merged...
> 
> Combat Tweak
> 
> * balance: Ash grasp blinds instead of knockback
> * Heretics marks area work differently; marks are applied by grasp, and triggered with the blade
> 
> Tweaks to Gamemode in general
> 
> * tweak: Rust spread datum is revised/optimized
> * tweak: Tears in reality no longer inflict brain damage when analyzed, instead just give a bad moodlet
> * tweak: Revised reality tears generation
> * tweak: Tears in reality appear after a time, and slowly disappear after 15 minutes
> * balance: Heretic effects (runes, tears) are invisible to silicons
> * spellcheck: Names and descriptions have been revised, thanks to @EOBGames
> 
> Heart Changes
> 
> * Heart now offers a pick between 3 targets
> * Doubled the charges sacrifices offer

It should be fine, and I don't see a reason why. All these changes are on TG.

File: code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm [hunk 17]
```diff
@@ -260,28 +270,35 @@
...
 				break
 
 		if(!LH.target)
 			var/datum/objective/A = new
 			A.owner = user.mind
-			var/datum/mind/targeted =  A.find_target()//easy way, i dont feel like copy pasting that entire block of code
-			LH.target = targeted.current
+			var/list/targets = list()
+			for(var/i in 0 to 3)
```
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm]:
```suggestion
			for(var/i in 1 to 3)
```

Do you want to pick 4 targets?

**Review by park66665 — DISMISSED** [review]
Okay, the delay is slowly driving me mad. I'll put this into a vote in order to stimulate discussion with AUTH48: I *will* merge this PR in 48 hours unless another maintainer objects. The author is free to make minuscule last minute changes.

**Review by park66665 — APPROVED** [review]
The time has come, and the wiki is updated. This way to merging PRs isn't very ideal, but large PRs also should be merged some day, and we have to make that some day happen even if the stars fail to align.

All clear. This PR is good to go.

**Comment by eeSPee** [comment]
Thanks

**Pull request merged**